src/Oscoin/API/HTTP/Handlers.hs

```diff
@@ -69,26 +69,19 @@ submitTransaction
        , Serialise (Crypto.Signature c)
        , Crypto.HasHashing c
        , Buildable (Crypto.Hash c)
-       , Crypto.HasDigitalSignature c
        )
     => ApiAction c s i a
 submitTransaction = do
-    tx <- getVerifiedTxBody
-    receipt <- node $ do
-        store <- view telemetryStoreL
-        Mempool.addTxs [tx]
-        forM_ [ TxsAddedToMempoolEvent [Crypto.fromHashed $ Crypto.hash @c tx]
-              , TxSubmittedEvent (Crypto.hash @c tx)] $
-            liftIO . Telemetry.emit store
-        pure $ TxSubmitResponse (hash @c tx)
-
-    respond accepted202 $ Ok receipt
-  where
-    getVerifiedTxBody = do
-        tx <- getBody
-        if verifyTx tx
-        then pure tx
-        else respond badRequest400 $ errBody "Invalid transaction signature"
+    tx <- getBody
+    addResult <- node $ Mempool.addTx tx
+    case addResult of
+        Left _ -> respond badRequest400 $ errBody "Invalid transaction signature"
+        Right _ -> do
+            telemetryStore <- node $ view telemetryStoreL
+            forM_ [ TxsAddedToMempoolEvent [Crypto.fromHashed $ Crypto.hash @c tx]
+                  , TxSubmittedEvent (Crypto.hash @c tx)] $
```

Comment:
I wonder if we should trigger a `TxSubmittedEvent` on failed validation too? @adinapoli-mndc

Reply:
My inclination would be to say 'yes', alongside with the relevant "validation failed" event. We can later tune down the verbosity of telemetry if that turns out to generate too much traffic 😉

Reply:
https://github.com/oscoin/oscoin/pull/555/commits/03cd7f8d81783a06a3a24bf3b90ffea3fb75fcb7